aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp

```diff
@@ -8,6 +8,32 @@
 namespace at {
 namespace native {
 
+Tensor NestedTensor_linear(
+    const Tensor& input,
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias_opt) {
+  auto* nt_input = get_nested_tensor_impl_or_null(input);
+  TORCH_CHECK(nested_tensor_impl_is_contiguous(nt_input));
+  TORCH_CHECK(input.dim() == 3 && weight.dim() == 2);
+  const auto last_dim = get_consistent_last_dim_of_nested_tensor(*nt_input);
+  TORCH_CHECK(
+      last_dim == weight.size(1),
+      "Shape mismatch for NestedTensor linear. NestedTensor last_dim: ",
+      last_dim,
+      " vs. dim 1 of rhs: ",
+      weight.size(1));
```

Comment:
I like how informative the error message is, but just to match the format across the code base I might reword to something like:

`Shape mismatch for NestedTensor linear: Expected input's (a nested tensor) 'last_dim' to equal 'weight.size(1)', but got: last_dim = x, and weight.size(1) = y`